### PR TITLE
Remove CustomerVoucher creation from draftOrderCreate

### PR DESCRIPTION
**Pull request by michal-macioszczyk**
I want to merge this change because I want to remove CustomerVoucher creation as the user might change (we should create VoucherCustomer during draftOrderComplete - will be done in https://github.com/saleor/saleor/issues/14382)

https://github.com/saleor/saleor/issues/14376

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/discount/utils.py [hunk 1]
```diff
@@ -59,8 +59,6 @@ def increase_voucher_usage(
 ) -> None:
     if voucher.usage_limit:
         increase_voucher_code_usage_value(code)
-    if voucher.apply_once_per_customer:
-        add_voucher_usage_by_customer(code, customer_email)
```
Comment (IKarbowiak) [inline saleor/discount/utils.py:63]: We cannot remove it from this function as the function is used in different places as well.

**Review by IKarbowiak — CHANGES_REQUESTED** [review]

File: saleor/discount/utils.py (hunk 1 above)
Comment (michal-macioszczyk) [reply saleor/discount/utils.py]: changed to new solution.

File: saleor/discount/utils.py [hunk 2]
```diff
@@ -55,11 +55,14 @@
 
 
 def increase_voucher_usage(
-    voucher: "Voucher", code: "VoucherCode", customer_email: str
+    voucher: "Voucher",
+    code: "VoucherCode",
+    customer_email: str,
+    allow_customer_voucher_creation: bool = True,
```
Comment (IKarbowiak) [inline saleor/discount/utils.py:61]: I would suggest changing the name to `increase_voucher_customer_usage` or smth like this, as we're creating the `VoucherCustomer` under the hood but it's not visible in this function

Comment (michal-macioszczyk) [reply saleor/discount/utils.py]: done

**Review by SzymJ — APPROVED** [review]

**Pull request merged**